lib/os/Kconfig

```diff
@@ -21,4 +21,23 @@ config BASE64
 	help
 	  Enable base64 encoding and decoding functionality
 
+config EVENTFD
+	bool "Enable support for eventfd"
+	depends on NET_BUF
```

Comment:
Note that how things are now in this PR, it cannot be used with the POSIX arch. So I'd ask either:
* To add here a `depends on !ARCH_POSIX` to avoid users getting hit by it by surprise.
OR
* Have the feature be compatible with it, by either renaming the conflicting types and symbols (`eventfd_t` , `eventfd_read` , `eventfd_write`) in the code or adding them to `posix_cheats.h` , and renaming the header or having the POSIX arch underlying OS include paths and the zephyr app/os path not be added to each other library when compiling.

Reply:
> Another option I guess would be to make the call behave the same way as in Linux and keeping the name?

That's the preferred way, in this in similar cases.

Reply:
I've added the symbols to `posix_cheats.h` and all tests now pass for `native_posix`.

Reply:
>all tests now pass for native_posix

Uhm, how are you building and running that test?
`lib/posix/eventfd.c` should only be compiled-in if POSIX_API is set. But POSIX_API won't be set when using the POSIX arch.
So if you build your test just targetting native_posix, you will be linking against the Linux `eventfd*` calls instead of the ones you are adding here.
Meaning, you wouldn't be testing `lib/posix/eventfd.c`.

For me that is fine, but I would then maybe blacklist the test in the posix arch targets? (unless your intention is to test that the API is equal to Linux, and that therefore the test passed also on Linux).

Reply:
Oh, I actually thought that was the idea :p. I mean to use the native calls when available... Correct that `lib/posix/eventfd.c` isn't tested on `native_posix` but it tests for same behaviour.

Reply:
> Oh, I actually thought that was the idea :p 

The idea as in "expecting somebody writing a Zephyr app, would want to have that app call the Linux eventfd"? (That would not produce sensible results from the application point of view. A Zephyr app running in native_posix should never be able to call the Linux API directly)

The only calls in native_posix that should be "native" from the app point of view are the ones to the libC utility functions (say `strcmp`), but none of the kernel calls (that's why posix_cheats.h was created, to rename those calls out of the way)

As it is now, I think this will confuse people into thinking this test is testing `lib/posix/eventfd.c`  when run in native_posix, even that it does not.
So I'd blacklist the test for the posix arch, (and probably also add a depends on !ARCH_POSIX on the EVENTFD config)

Reply:
> As it is now, I think this will confuse people into thinking this test is testing `lib/posix/eventfd.c` when run in native_posix, even that it does not.
> So I'd blacklist the test for the posix arch, (and probably also add a depends on !ARCH_POSIX on the EVENTFD config)

I have excluded posix arch from test and added the !ARCH_POSIX depends.